src/main/java/nicole/task/Task.java

```diff
@@ -0,0 +1,91 @@
+package nicole.task;
+
+import java.time.LocalDate;
+
+import nicole.nicoleexceptions.NicoleException;
+
+public class Task {
+    private boolean taskCompleted;
+    private String name;
+
+    /**
+     * Initialises a Task. The default state of a Task is incomplete.
+     *
+     */
+    public Task() {
+        this.taskCompleted = false;
+    }
+
+    protected void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Factory method for Tasks.
+     *
+     * @param name the user request.
+     * @param taskType 'T' for Todo, 'D' for Deadline, 'E' for Event.
+     * @throws NicoleException if any of Task, Deadline, or Event throws it due to
+     *                         invalid request.
+     */
+    public static Task taskFactory(String name, char taskType) throws NicoleException {
+        if (taskType == 'T') {
```

Comment:
You could use a switch statement! This would then allow unknown `taskType` values that weren't captured in your cases to have a default handler flow.